### PR TITLE
upgraded tsup to generate .d.mts files for better TypeScript/ESM support

### PR DESCRIPTION
**Pull request by deathcrafter**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,12 +28,14 @@
...
   },
   "dependencies": {
-    "fflate": "^0.7.4"
+    "fflate": "^0.8.2"
   },
   "devDependencies": {
-    "prettier": "^2.7.1",
-    "tsup": "^6.2.3",
-    "typescript": "^4.8.4"
+    "@swc/core": "^1.4.8",
```
Comment (reslear) [inline package.json]: Why do we need `@swc/core` then?

File: tsup.config.ts [hunk 2]
```diff
@@ -1,13 +1,12 @@
 import { defineConfig } from 'tsup'
+import { copyFileSync } from 'fs'
```
Comment (reslear) [inline tsup.config.ts]: Why do we need that line ?

**Review by reslear — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (deathcrafter) [reply package.json]: image

We don't necessarily need it, but it was a warning by `tsup` so I thought might as well include it.

File: tsup.config.ts (hunk 2 above)
Comment (deathcrafter) [reply tsup.config.ts]: We don't actually, it is a mistake.

File: package.json (hunk 1 above)
Comment (reslear) [reply package.json]: ok, i delete emitDecoratorMetadata

**Review by reslear — APPROVED** [review]

**Comment by reslear** [comment]
Thank you. You can try it.

**Pull request merged**